.github/workflows/build-api.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build API Module
+run-name: Build API Module
+
+on:
+  pull_request:
+    paths:
+      - 'api/**.go'
+      - 'api/go.mod'
+      - 'api/go.sum'
+  merge_group:
+    paths:
+      - 'api/**.go'
+      - 'api/go.mod'
+      - 'api/go.sum'
+
+jobs:
+  build:
+    name: Build API
+    if: ${{ !startsWith(github.head_ref, 'dependabot/') }}
```

Comment:
What if a dependabot update makes `api` unbuildable?

Reply:
That would be caught by the regular test and build jobs that we already have.